xtevent/_eventols.ado

```diff
@@ -192,6 +249,50 @@ program define _eventols, rclass
 		
 	******** SA *******
 	if "`sun_abraham'"!=""{
+
+		* Another consistency check: obs with cohort should have some treated values, obs with control_cohort should have some untreated values
+		* Not checking that all cohorts have treated values, not checking that z is always zero for never treat, so these are minimal checks
+
+		if "`cohorttype'"=="" {
+			qui levelsof z if `cohortvar'!=. & `touse'
+			if r(levels)=="0" {
+				di as err _n "Treated observations according to cohort variable `cohortvar' are inconsistent"
+				di as err "with values of the policy variable `z'"
+				exit 301
+			}
+
+			if "`control_cohorttype'"=="variable" {
+				qui count if `z'==0 & `control_cohortvar' & `touse'
+				if r(N)==0 {
+					di as err _n "Untreated observations according to control cohort variable `control_cohortvar'"
+					di as err "are inconsistent with values of the policy variable `z'"
+				}
+			}
+		}
+
+		* Create the cohort variable if requested
+		if "`cohorttype'"=="create" {
+			cap assert `binnorev'==1
+			if _rc {
+				di as err _n "The policy variable is not binary or treatment reverts"
+				di as err "Cannot create treatment cohort variables"
+				exit 301
+			}
+
+			tempvar timet 
+			qui gen `timet'=`timevar' if `policyvar'==1 & `touse'
+			qui by `panelvar' : egen _cohort = min(`timet')
+			loc cohortvar "_cohort"			
+		}
+
+		* If control_cohort is missing, take the values with cohort = .
+		if "`cohortvar'"!="" & "`control_cohorttype'"=="create" {
+			if "`control_cohort'"=="" di as txt _n "Control cohort not specified. Using values with cohort variable == . as the control cohort"
+			else di as txt _n "Using values with cohort variable == . as the control cohort"
+			gen _control_cohort = (`cohortvar' == .)

```

Comment:
@jorpppp 
Maybe we can add a check for an existing variable `_control_cohort`.

Reply:
Solved in 
https://github.com/JMSLab/xtevent/pull/185/commits/474dd84512cd7a8d4b8321fb7c8e94a629ff495d